crates/e2e/tests/e2e/partially_fillable_observed_score.rs

```diff
@@ -145,17 +145,24 @@ async fn test(web3: Web3) {
         .await
         .unwrap();
 
-    let competition_0 = services
-        .get_solver_competition(trades[0].tx_hash.unwrap())
-        .await
-        .unwrap();
-    let competition_1 = services
-        .get_solver_competition(trades[1].tx_hash.unwrap())
-        .await
-        .unwrap();
-
+    let mut competitions: Vec<SolverCompetitionAPI> = vec![
+        services
+            .get_solver_competition(trades[0].tx_hash.unwrap())
+            .await
+            .unwrap(),
+        services
+            .get_solver_competition(trades[1].tx_hash.unwrap())
+            .await
+            .unwrap(),
+    ];
+    competitions.sort_by_key(|comp| comp.auction_id);
+    let competition_0 = competitions[0].clone();
+    let competition_1 = competitions[1].clone();
```

Comment:
nanonit. I believe given that the length is known, we'd normally use a slice instead of a Vec and unpack directly instead of clone (doesn't make a big difference here but is more "idiomatic"):

```suggestion
    let mut competitions = [
        services
            .get_solver_competition(trades[0].tx_hash.unwrap())
            .await
            .unwrap(),
        services
            .get_solver_competition(trades[1].tx_hash.unwrap())
            .await
            .unwrap(),
    ];
    competitions.sort_by_key(|comp| comp.auction_id);
    let [competition_0, competition_1] = competitions;
```